flycheck.el

```diff
@@ -1945,9 +1970,26 @@ nil otherwise."
                                 (funcall enabled)))))
     (if shall-enable
         (cl-pushnew checker flycheck-enabled-checkers)
-      (cl-pushnew checker flycheck-disabled-checkers))
+      (cl-pushnew checker flycheck-automatically-disabled-checkers))
     shall-enable))
 
+(defun flycheck-reset-enabled-checker (checker)
+  "Reset the `:enabled' test of CHECKER.
+
+Forget that CHECKER has been enabled or automatically disabled
+from a previous `:enabled' test.  Once a checker has been enabled
+or automatically disabled, `flycheck-may-enable-checker' will
+always be constant (t or nil respectively).
+
+If you wish to test the `:enabled' predicate again, you must
+first reset its state using this function."
```

Comment:
I think this description is a bit too technical.  Maybe just something along the lines of `clear Flycheck's cache of :enabled properties`?

EDIT: Oh, except I see that this removes the value from `flycheck-disabled-checkers`, too.  But see below